Mathlib/RingTheory/LocalRing/MaximalIdeal/Basic.lean

```diff
@@ -99,3 +99,40 @@ end LocalRing
 
 theorem LocalRing.maximalIdeal_eq_bot {R : Type*} [Field R] : LocalRing.maximalIdeal R = ⊥ :=
   LocalRing.isField_iff_maximalIdeal_eq.mp (Field.toIsField R)
+
+section Nilrad_max_localization
+
+open Ideal
+
+variable {R : Type*} [CommSemiring R] {S : Type*} [CommSemiring S] [Algebra R S] {M : Submonoid R}
+
+/--
+An element `x` of a commutative local semiring is not contained in the maximal ideal
+iff it is a unit.
+-/
+theorem LocalRing.not_mem_maximalIdeal [LocalRing R] (x : R) :
+    x ∉ LocalRing.maximalIdeal R ↔ IsUnit x := by
+  simp only [mem_maximalIdeal, mem_nonunits_iff, not_not]
+
+/--
+Let `S` be the localization of a commutative semiring `R` at a submonoid `M` that does not
+contain 0. If the nilradical of `R` is maximal then there is a ring isomorphism between `R` and `S`.
+-/
+noncomputable def nilradmax_localization_IsSelf (h : (nilradical R).IsMaximal) (h' : (0 : R) ∉ M)
```

Comment:
The naming convention requires `def`s to be in `lowerCamelCase`, i.e., `nilradMaxLocalizationIsSelf` in this case.